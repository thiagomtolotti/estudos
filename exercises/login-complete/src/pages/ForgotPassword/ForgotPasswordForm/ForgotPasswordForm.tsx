import { useForgotPassword } from "@/hooks/useForgotPassword";
import { FormEvent, useRef } from "react";
// import styles from "./ForgotPasswordForm.module.css";

interface ForgotPasswordFormProps {
	submit: (email: string) => void;
}

export const ForgotPasswordForm = ({ submit }: ForgotPasswordFormProps) => {
	const emailRef = useRef<HTMLInputElement>(null);

	function handleSubmit(ev: FormEvent) {
		ev.preventDefault();

		if (!emailRef.current?.value) {
			console.error("Insira um endere√ßo de email");
			return;
		}

		submit(emailRef.current.value);
	}

	return (
		<form onSubmit={handleSubmit}>
			<input type="email" placeholder="email" ref={emailRef} />

			<button>Enviar</button>
		</form>
	);
};
