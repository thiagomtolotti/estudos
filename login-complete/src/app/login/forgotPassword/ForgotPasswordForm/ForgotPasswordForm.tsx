import { FormEvent, useRef } from "react";
// import styles from "./ForgotPasswordForm.module.css";

interface ForgotPasswordFormProps {
  onSubmit(email: string): void;
}

export const ForgotPasswordForm = ({ onSubmit }: ForgotPasswordFormProps) => {
  const emailRef = useRef<HTMLInputElement>(null);

  function handleSubmit(ev: FormEvent) {
    ev.preventDefault();

    if (!emailRef.current?.value) {
      console.error("Insira um endere√ßo de email");
      return;
    }

    onSubmit(emailRef.current.value);
  }

  return (
    <form onSubmit={handleSubmit}>
      <input type="email" placeholder="email" ref={emailRef} />

      <button>Enviar</button>
    </form>
  );
};
