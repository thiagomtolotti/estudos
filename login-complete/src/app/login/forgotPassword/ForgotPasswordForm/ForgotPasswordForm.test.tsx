import {
  RenderResult,
  fireEvent,
  render,
  screen,
} from "@testing-library/react";
import ForgotPasswordForm from ".";

console.error = jest.fn();
const mockSubmit = jest.fn();

describe("ForgotPasswordForm", () => {
  let emailInput: HTMLInputElement, sendButton: HTMLButtonElement;
  let container: RenderResult;

  beforeEach(() => {
    container = render(<ForgotPasswordForm onSubmit={mockSubmit} />);

    emailInput = screen.getByPlaceholderText("email");
    sendButton = screen.getByRole("button");
  });

  it("Should render correctly", () => {
    expect(container.container).toMatchSnapshot();
  });

  it("Should have a email input and a button", () => {
    expect(emailInput).toBeInTheDocument();
    expect(sendButton).toBeInTheDocument();
  });

  describe("the onSubmit function", () => {
    it("Should not be called if the email input is empty", () => {
      fireEvent.click(sendButton);

      expect(console.error).toHaveBeenCalledWith("Insira um endereÃ§o de email");
    });

    it("Shouldn't be called if the email is not filled correctly", () => {
      const testInput = "test";

      fireEvent.input(emailInput, { target: { value: testInput } });

      fireEvent.click(sendButton);

      expect(mockSubmit).not.toHaveBeenCalled();
    });

    it("Should be called if the email is filled correctly", () => {
      const testEmail = "test@test.com";

      fireEvent.input(emailInput, { target: { value: testEmail } });

      fireEvent.click(sendButton);

      expect(mockSubmit).toHaveBeenCalledWith(testEmail);
    });
  });
});
