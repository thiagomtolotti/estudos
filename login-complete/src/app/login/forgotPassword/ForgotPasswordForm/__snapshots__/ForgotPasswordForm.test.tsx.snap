// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordForm Should render correctly 1`] = `
<div>
  <form>
    <input
      placeholder="email"
      type="email"
    />
    <button>
      Enviar
    </button>
  </form>
</div>
`;
